GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -485,49 +482,64 @@ private void RequestRefresh()
 
-        private void ResetSelectedItemsTo(bool actsAsChild)
+        private void ResetSelectedItemsTo(bool actsAsChild, bool deleteUncommittedAddedItems)
         {
             var selectedItems = DiffFiles.SelectedItems.ToList();
             if (!selectedItems.Any())
             {
                 return;
             }
 
-            if (actsAsChild)
+            try
             {
-                // selected revisions
-                var deletedItems = selectedItems
-                    .Where(item => item.Item.IsDeleted)
-                    .Select(item => item.Item.Name).ToList();
-                Module.RemoveFiles(deletedItems, false);
-
-                foreach (var childId in selectedItems.SecondIds())
+                if (actsAsChild)
                 {
-                    var itemsToCheckout = selectedItems
-                        .Where(item => !item.Item.IsDeleted && item.SecondRevision.ObjectId == childId)
+                    // Reset to selected revision
+
+                    var deletedItems = selectedItems
+                        .Where(item => item.Item.IsDeleted)
                         .Select(item => item.Item.Name).ToList();
-                    Module.CheckoutFiles(itemsToCheckout, childId, force: false);
-                }
-            }
-            else
-            {
-                // acts as parent
-                // if file is new to the parent or is copied, it has to be removed
-                var addedItems = selectedItems
-                    .Where(item => item.Item.IsNew || item.Item.IsCopied)
-                    .Select(item => item.Item.Name).ToList();
-                Module.RemoveFiles(addedItems, false);
+                    Module.RemoveFiles(deletedItems, false);
 
-                foreach (var parentId in selectedItems.FirstIds())
+                    foreach (var childId in selectedItems.SecondIds())
+                    {
+                        var itemsToCheckout = selectedItems
+                            .Where(item => !item.Item.IsDeleted && item.SecondRevision.ObjectId == childId)
+                            .Select(item => item.Item.Name).ToList();
+                        Module.CheckoutFiles(itemsToCheckout, childId, force: false);
+                    }
+                }
+                else
                 {
-                    var itemsToCheckout = selectedItems
-                        .Where(item => !item.Item.IsNew && item.FirstRevision?.ObjectId == parentId)
-                        .Select(item => item.Item.Name).ToList();
-                    Module.CheckoutFiles(itemsToCheckout, parentId, force: false);
+                    // Reset to parent revision
+
+                    // If file is new to the parent or is copied, it has to be deleted or removed if un/committed, respectively
+                    IEnumerable<FileStatusItem> addedItems = selectedItems.Where(item => item.Item.IsAdded);
+                    if (selectedItems.First().Item.IsUncommitted)
```

Comment:
`selectedItems.First()` would throw if the enumeration was empty.

Reply:
Would but cannot. There is an `if (!any) return` at the entry of the function.

Reply:
Nope, selected is either the WorkTree *or* the Index *or* a real commit. All files of a diff belong to the same artificial or real commit.

Reply:
Ok the way it reads it seems like each value in selecteditems is one file.